### ISSUE TITLE
Slightly confusing error message

### ISSUE THREAD

**Issue by aezarebski** [ekmett/either#66]
https://github.com/ekmett/either/blob/544fd839f83386ae13f1ec1514618ec226a69489/src/Data/Either/Combinators.hs#L111

The `fromLeft'` and `fromRight'` functions use an error message that had me scratching my head for a while because I used both `fromRight` and `fromRight'` in the same module. Currently it is

```
"Data.Either.Combinators.fromRight: Argument takes form 'Left _'"
```

I think it would be clearer to have 

```
"Data.Either.Combinators.fromRight' encountered a value of form 'Left _', consider using Data.Either.Combinators.fromRight and providing a default value."
```

This is highly subjective though...

Thanks for the great library!

**Comment by RyanGlScott** [comment]
Indeed, that should definitely say `Data.Either.Combinators.fromRight'` instead of `Data.Either.Combinators.fromRight` at minimum. I think mentioning `fromRight` as an alternative is also a good idea. Would you care to submit a patch?

**Comment by aezarebski** [comment]
Thanks, have done so.

**Comment by RyanGlScott** [comment]
Fixed in #67 / ddd1e9427ad9b2e598f277472aff5262a6b96c9e.
